src/NLog/LayoutRenderers/ExceptionLayoutRenderer.cs

```diff
@@ -177,6 +178,16 @@ public List<ExceptionRenderingFormat> InnerFormats
             private set;
         }
 
+        /// <summary>
+        /// Registers a custom exception renderer type
+        /// </summary>
+        /// <typeparam name="T">Type of the exception</typeparam>
+        /// <param name="renderer">The renderer for converting exception to string</param>
+        public static void RegisterExceptionRenderer<T>(Func<T, string> renderer) where T : Exception
+        {
+            _customRenderingfunctions.Add(typeof(T), exception => renderer((T) exception));
```

Comment:
`_customRenderingfunctions[type] = ` is maybe better (no exception when duplicate)

Reply:
Done